WikiWatershed/model-my-watershed
# Split up & speed up  Mapshed job setup

This addresses a couple issues with the setup process for Mapshed jobs. First, it breaks up the setup task into smaller tasks, which, in general, are easier to debug and profile both for purposes of this issue as well as in future. Second, it makes some minor changes to the implementation with the aim of speeding the process up in general, primarily by removing a notable blocking method invocation as well as parallelizing the process where possible.

To test:
* clone this branch
* it may be worth changing `Vagrantfile` to bump up the settings for `worker`; I was testing it with 2 CPUs and 2048 GB memory
* bring up the environment
* go through typical user actions with the app, an ensure that the app behavior is consistent with expectations

The `POST` to `/api/modeling/start/mapshed` should be faster than it was before or currently is on staging (the call in staging should take 18-20 seconds, this implementation could take 2-8 seconds depending on the VM resources). The overall model run was also faster in my testing, but there are several mitigating factors if a local version is compared to staging, so it's probably not worth spending much time evaluating.

Connects #1491
